core/src/main/java/org/infinispan/interceptors/impl/TransactionalExceptionEvictionInterceptor.java

```diff
@@ -221,7 +221,9 @@ public Object visitPrepareCommand(TxInvocationContext ctx, PrepareCommand comman
          throw log.containerFull(maxSize);
       }
 
-      pendingSize.put(ctx.getGlobalTransaction(), changeAmount);
+      if (!command.isOnePhaseCommit()) {
```

Comment:
This seems like a separate issue, we should probably fix this in 9.4 as well.

Reply:
Good point, I created ISPN-10579